### PR TITLE
Use `:nth-child/:nth-last-child` selector to set frozen and summary row styles

### PR DESCRIPTION
**Pull request by amanmahajan7**

### CODE REVIEW

File: src/GroupRow.tsx [hunk 1]
```diff
@@ -16,7 +16,8 @@ const groupRow = css`
       background-color: var(--rdg-header-background-color);
     }
 
-    > .${cell}:not(:last-child):not(.${cellFrozenLast}) {
+    > .${cell}:not(:last-child):not(.${cellFrozen}),
```
Comment (nstepien) [inline src/GroupRow.tsx:19]: https://developer.mozilla.org/en-US/docs/Web/CSS/:not#description
>You can negate several selectors at the same time. Example: `:not(.foo, .bar)` is equivalent to `:not(.foo):not(.bar)`.

```suggestion
    > .${cell}:not(:last-child, .${cellFrozen}),
```

File: src/style/cell.ts [hunk 2]
```diff
@@ -35,15 +35,12 @@ export const cellFrozen = css`
...
-  }
-`;
-
-export const cellFrozenClassname = `rdg-cell-frozen ${cellFrozen}`;
 
-export const cellFrozenLast = css`
-  @layer rdg.Cell {
-    box-shadow: calc(2px * var(--rdg-sign)) 0 5px -2px rgba(136, 136, 136, 0.3);
+    /* Add box-shadow on the last frozen cell */
+    &:has(+ :not(&)) {
```
Comment (nstepien) [inline src/style/cell.ts:40]: Does this work if the last frozen cell is also the last cell in the row?

File: src/style/core.ts [hunk 3]
```diff
@@ -88,6 +89,18 @@ const root = css`
         ${darkTheme}
       }
     }
+
+    :nth-last-child(1 of .${topSummaryRowClassname}) {
```
Comment (nstepien) [inline src/style/core.ts:93]: Can we specify `>`?
```suggestion
    > :nth-last-child(1 of .${topSummaryRowClassname}) {
```

File: src/style/core.ts [hunk 4]
```diff
@@ -88,6 +89,18 @@ const root = css`
...
+      > .${cell} {
+        border-block-end: 2px solid var(--rdg-summary-border-color);
+      }
+    }
+
+    :nth-child(1 of .${bottomSummaryRowClassname}) {
```
Comment (nstepien) [inline src/style/core.ts:99]:
```suggestion
    > :nth-child(1 of .${bottomSummaryRowClassname}) {
```

File: src/style/cell.ts (hunk 2 above)
Comment (amanmahajan7) [reply src/style/cell.ts]: It did not work. Fixed with `:nth-last-child`

File: src/style/core.ts [hunk 5]
```diff
@@ -88,6 +89,23 @@ const root = css`
         ${darkTheme}
       }
     }
+
+    /* Add box-shadow on the last frozen cell */
+    :nth-last-child(1 of .${cellFrozen}) {
```
Comment (amanmahajan7) [inline src/style/core.ts:94]: Added here as header does not use `row` class. I can instead add it to both `row` and `header-row` if you prefer

Comment (amanmahajan7) [reply src/style/core.ts]: Added a variable instead. I think it is better to use more specific styles

File: src/style/row.ts [hunk 6]
```diff
@@ -17,6 +19,11 @@ export const row = css`
         background-color: var(--rdg-row-selected-hover-background-color);
       }
     }
+
+    /* Add box-shadow on the last frozen cell */
+    > :nth-last-child(1 of .${cellFrozen}) {
```
Comment (nstepien) [inline src/style/row.ts:24]: Nice!

File: src/style/core.ts [hunk 7]
```diff
@@ -29,6 +31,7 @@ const darkTheme = `
   --rdg-row-hover-background-color: hsl(0deg 0% 9%);
   --rdg-row-selected-background-color: hsl(207deg 76% 42%);
   --rdg-row-selected-hover-background-color: hsl(207deg 76% 38%);
+  --rdg-cell-frozen-box-shadow: calc(2px * var(--rdg-sign)) 0 5px -2px rgba(136, 136, 136, 0.3);
```
Comment (nstepien) [inline src/style/core.ts:34]: We don't need to define this twice if it has the same value. We can move it next to `--rdg-selection-color` and `--rdg-font-size`

File: src/style/row.ts [hunk 8]
```diff
@@ -17,6 +19,11 @@ export const row = css`
...
+      box-shadow: var(--rdg-cell-frozen-box-shadow);
+    }
```
Comment (nstepien) [inline src/style/row.ts:26]: Does this need to be part of the (Header)Row styles?
We can move it to cell styles, like
```css
&:nth-... {}
```

File: src/GroupRow.tsx [hunk 9]
```diff
@@ -16,7 +16,8 @@ const groupRow = css`
       background-color: var(--rdg-header-background-color);
     }
 
-    > .${cell}:not(:last-child):not(.${cellFrozenLast}) {
+    > .${cell}:not(:last-child, .${cellFrozen}),
+    > .${cellFrozen}:has(+ .${cellFrozen}) {
```
Comment (nstepien) [inline src/GroupRow.tsx:20]: Does this work?
```suggestion
    > :nth-last-child(n + 1 of .${cellFrozen}) {
```

Comment (amanmahajan7) [reply src/GroupRow.tsx:20]: Does not work

Comment (nstepien) [reply src/GroupRow.tsx:20]: `n + 2` then? 🤔

File: src/style/row.ts (hunk 8 above)
Comment (amanmahajan7) [reply src/style/row.ts]:
```
&:nth-last-child(1 of &) {
    box-shadow: var(--rdg-cell-frozen-box-shadow);
 }
```
This works

File: src/GroupRow.tsx (hunk 9 above)
Comment (amanmahajan7) [reply src/GroupRow.tsx]: This works. How does it work :) ?

Comment (nstepien) [reply src/GroupRow.tsx]: For all the `nth` selectors, you can use `n` with some arithmetic to select specific elements.
https://developer.mozilla.org/en-US/docs/Web/CSS/:nth-child
I don't know why `+ 2` is needed to skip the first match though.

**Review by nstepien — APPROVED** [review]
:shipit:

**Pull request merged**